transcendence-app/src/chat/chat.gateway.ts

```diff
@@ -30,10 +32,45 @@ export class ChatGateway {
   constructor(
     private chatService: ChatService,
     private chatroomMemberService: ChatroomMemberService,
+    private userService: UserService,
   ) {}
 
+  @SubscribeMessage('chatMessage')
+  async handleChatMessage(
+    @MessageBody()
+    createChatMessageDto: CreateChatMessageDto,
+    @ConnectedSocket() client: Socket,
+  ) {
+    const { userId: recipientId, content } = createChatMessageDto;
+
+    const senderId = client.request.user.id;
+
+    const block = await this.userService.getBlockRelation(
+      recipientId,
+      senderId,
+    );
+
+    if (block?.isUserBlockedByMe || block?.amIBlockedByUser) {
+      throw new WsException('There is an existing blockage. Forbidden.');
+    }
+
+    const message = await this.chatService.saveOneToOneChatMessage({
+      senderId,
+      recipientId,
+      content,
+    });
+
+    if (message) {
+      this.server.to(recipientId).emit('chatMessage', { ...message });
+    } else {
```

Comment:
Yes, but from the user perspective it couldn't be sent because no user will receive it unless is saved.

Maybe as an internal log for us could be very useful to log the db error so we knew what happened.